### ISSUE TITLE
Symbol not found after update to the last version on nuget

### ISSUE THREAD

**Issue by JFDionne** [Flash3001/Xamarin.Swift#21]
Hi, I just upgrade to the last version becuase of the invalid swift we refuse from apple, that been fixed, but now I have this error when I start my app:

Termination Description: DYLD, Symbol not found: __T0SSN | 
Referenced from: /private/var/containers/Bundle/Application/5C604E0F-FF5F-4413-A4D0-36731FA94CF2/SIV_Lite.iOS.app/Frameworks/Polyline.framework/Polyline | 
Expected in: /private/var/containers/Bundle/Application/5C604E0F-FF5F-4413-A4D0-36731FA94CF2/SIV_Lite.iOS.app/Frameworks/libswiftCore.dylib | 
in /private/var/containers/Bundle/Application/5C604E0F-FF5F-4413-A4D0-36731FA94CF2/SIV_Lite.iOS.app/Frameworks/Polyline.framework/Polyline

Did you removed Symbol from your last version?

**Comment by Flash3001** [comment]
Hey @JFDionne .

That happens when the library you are using was not compiled using the latest Xcode version.

When you add the SwiftSupport library it is just adding the Swift runtime to your app. So when you have a library that was built against Swift 4.0 or Swift 2.0 it shouldn't work on a 4.1.2 runtime. That kind of stuff breaks even on point releases. 

If you have access to Polyline.framework source code, just open it on Xcode and recompile and redo the binding. If it is not the case then you have to ask the library maintainer to do it. Otherwise, the solution is to downgrade Xcode and SwiftSupport to the previous version you were using.

All 3 needs to upgrade together: Library Swift version = SwiftSupport version = Xcode on the build server version.

Before updating you had the library and SwiftSupport on the same version, but Xcode was newer. So the package_ipa Script would cause your app to have one version of the Swift runtime on the MyApp.api/SwiftSupport folder and another version in MyApp.api/Payload/MyApp.app/Frameworks folder.

After the update the 2 matches, but not the library itself.

**Comment by JFDionne** [comment]
Oh, I see thank.

Any chance you know which version of xcode can work with the "old" version?

**Comment by Flash3001** [comment]
Sure. You can check that on the Release Notes for each Nuget package.

https://www.nuget.org/packages/Xamarin.Swift4.Core/4.2.1.1

On this link you can find the **History Version**. On it the first 3 number match the exact Swift version it was built for and the Release notes says which is the version of Xcode you want. 

Let's say you pick https://www.nuget.org/packages/Xamarin.Swift4.Core/4.1.2

4.1.2 = Xcode 9.4

https://www.nuget.org/packages/Xamarin.Swift4.Core/4.2.0

4.2.0 = Xcode 10
4.2.1.1 = Xcode 10.1

**Comment by JFDionne** [comment]
Thanks !

**Comment by andrew-nemtsev** [comment]
xcode 10.2 already released (and referenced issue were solved by installing of 10.1).
Can we expect release of Xamarin.Swift4 4.2.1.2 ?

**Comment by Flash3001** [comment]
@andrew-nemtsev Xcode 10.2 brings support for Swift 5. 

The current release strategy for this project up to this point was for each major Swift release I would wrap each individual dylib into a Nuget package and release it + 1 MSBuild script. (total of 43 packages for Swift 4) and for each Xcode/Swift update a new minor version for the 42 packages. It was working so far, but Swift 5 brought some changes and I wanted to change how the MSBuild script works. The new package will be: 

- a single Nuget package (no need to download a bunch of heavy packages)
- works for iOS with experimental support for macOS and tvOS
- any version of Xcode/Swift should work with the same script (the script will check if your Xcode version is compatible with your libraries) so this project doesn't need to be updated when a new Xcode is released (libraries built before Swift 5 will have to).
- it will remove architectures from the Swift dylibs your app doesn't need.
- it will include the SwiftSupport folder on the IPA during the normal build, no need to run the package_ipa.sh script.
- if the library is built using Swift 5 and your app has a minimum target of iOS 12.2 then no Swift runtime library is included as iOS 12.2 already comes with it.

All the code is already in place, but I need to do more testing before pushing it to Nuget (you can check it here https://github.com/Flash3001/Xamarin.SwiftSupport/tree/master/SwiftSupport)

**Comment by Kingamattack** [comment]
Hi, I'm facing the same issue than @andrew-nemtsev and @JFDionne assuming:
```
Termination Reason:    DYLD, [0x4] Symbol missing

Application Specific Information:
dyld: launch, loading dependent libraries
DYLD_FALLBACK_LIBRARY_PATH=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/Library/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/usr/lib
DYLD_ROOT_PATH=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/Library/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot
DYLD_FALLBACK_FRAMEWORK_PATH=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/Library/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/System/Library/Frameworks

Dyld Error Message:
  Symbol not found: _$s12PSOperations9OperationC15instanceContext33_1C9B4685185CCDD77C246BA30CDDA45FLLSivM
  Referenced from: /Users/USER/Library/Developer/CoreSimulator/Devices/8A0EF93A-CB45-457C-B060-4540C2D339B4/data/Containers/Bundle/Application/11B65A74-3C6F-4A20-AFDA-E3923339E424/SampleAppiOS.app/Frameworks/PackIdLib.framework/PackIdLib
  Expected in: /Users/USER/Library/Developer/CoreSimulator/Devices/8A0EF93A-CB45-457C-B060-4540C2D339B4/data/Containers/Bundle/Application/11B65A74-3C6F-4A20-AFDA-E3923339E424/SampleAppiOS.app/Frameworks/PSOperations.framework/PSOperations
 in /Users/USER/Library/Developer/CoreSimulator/Devices/8A0EF93A-CB45-457C-B060-4540C2D339B4/data/Containers/Bundle/Application/11B65A74-3C6F-4A20-AFDA-E3923339E424/SampleAppiOS.app/Frameworks/PackIdLib.framework/PackIdLib
```

Here is what I got in the Console app when I try to debug on a simulator (iPhone X 12.2).
I'm note sure about the swift version used to build the lib (probably Swift 4.2 or 5) but according to your instructions, instead of adding a bunch of Nuget Packages in my iOS project, I only added this one Xamarin.Swift in its latest version but I still have this crash at launch when I try to call a method/object from my binding library.

Except put a reference to my binding project and install this Nuget Package, do I have to do something specific? I'm a bit confuse...
